sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -68,20 +72,43 @@ export class AppConfigurationClient {
    * Initializes a new instance of the AppConfigurationClient class.
    * @param connectionString Connection string needed for a client to connect to Azure.
    */
-  constructor(connectionString: string) {
-    const regexMatch = connectionString.match(ConnectionStringRegex);
-    if (regexMatch) {
-      const appConfigCredential = new AppConfigCredential(regexMatch[2], regexMatch[3]);
-      this.client = new AppConfiguration(appConfigCredential, apiVersion, {
-        baseUri: regexMatch[1],
+  constructor(connectionString: string);
+  /**
+   * Initializes a new instance of the AppConfigurationClient class using 
+   * a TokenCredential
+   * @param endpoint The endpoint of the App Configuration service (ex: https://sample.azconfig.io).
+   * @param tokenCredential A TokenCredential.
```

Comment:
```suggestion
   * @param tokenCredential An object that implements the `TokenCredential` interface used to authenticate requests to the service. Use the @azure/identity package to create a credential that suits your needs.
```
Or something like the above? (took from keyvalt-keys)

Also, in our other libraries we just call it `credential` instead of `tokenCredential`.